### PR TITLE
Modify event test

### PR DESCRIPTION
**Pull request by aamare1001**

### CODE REVIEW

File: __tests__/modifyEvent.test.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        json: () => Promise.resolve({}),
+      },
+    ])
+    const result = mockModifyEvent(ORIGINAL_EVENT)
+    await act(async () => {
+      await result.current.updateDuration(
+        MODIFIED_EVENT.duration.start,
+        MODIFIED_EVENT.duration.end,
+      )
+      await result.current.updateLocation(MODIFIED_EVENT.location)
```
Comment (Victoria-Williamson) [inline __tests__/modifyEvent.test.tsx:28]: these are not async functions they are hooks. Also since they are just updating a value in a hook you don't need to test it. Even if it was wrong we wouldn't be able to do anything since it should already be tested from react

File: __tests__/modifyEvent.test.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+describe("Modify Event Hook", () => {
+  beforeEach(() => {
+    jest.resetAllMocks()
+    // TODO: fix issue with react versions
+    jest.spyOn(console, "error").mockImplementation(() => {})
+    jest.spyOn(console, "log").mockImplementation(() => {})
+  })
+
+  test("Successfully modified", async () => {
```
Comment (Victoria-Williamson) [inline __tests__/modifyEvent.test.tsx:13]: Add a better explanation for what it is testing for. Succesfully modified when : ____

File: __tests__/modifyEvent.test.tsx (hunk 1 above)
Comment (Victoria-Williamson) [reply __tests__/modifyEvent.test.tsx:28]: Also, assuminig this test is just making sure successful modification when event isn't empty, you shouldn't have to do updateDescription, updateTitle, etc.

File: __tests__/modifyEvent.test.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        MODIFIED_EVENT.duration.end,
+      )
+      await result.current.updateLocation(MODIFIED_EVENT.location)
+      await result.current.updateDescription(MODIFIED_EVENT.description)
+      await result.current.updateTitle(MODIFIED_EVENT.title)
+    })
+
+    expect(await result.current.modify(() => {}, true)).toEqual(true)
+  })
+  test("Blank Title", async () => {
```
Comment (Victoria-Williamson) [inline __tests__/modifyEvent.test.tsx:35]: Make a better title for the test

File: __tests__/modifyEvent.test.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        json: () => Promise.resolve({}),
+      },
+    ])
+    const result = mockModifyEvent(ORIGINAL_EVENT)
+    await act(async () => {
+      await result.current.updateDuration(
+        MODIFIED_EVENT.duration.start,
+        MODIFIED_EVENT.duration.end,
+      )
+      await result.current.updateLocation(MODIFIED_EVENT.location)
```
Comment (Victoria-Williamson) [inline __tests__/modifyEvent.test.tsx:50]: If this one is just for updating an Event, with a missing value you could pass in the original event with one of the values empty

File: __tests__/modifyEvent.test.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      await result.current.updateDescription(MODIFIED_EVENT.description)
+      await result.current.updateTitle("")
+    })
+
+    expect(await result.current.modify(() => {}, true)).toEqual(false)
+  })
+
+  test("Blank Location", async () => {
```
Comment (Victoria-Williamson) [inline __tests__/modifyEvent.test.tsx:58]: This and the test above test for the same case, so I would suggest maybe making sure that the current test gets updated locally as well since that would cover mode lines of code

File: utility/hooks/trip.tsx [hunk 6]
```diff
@@ -364,18 +364,11 @@ export function TripProvider({ children }: { children: React.ReactNode }) {
...
     }
-
-    list[list.length - 1].push({
-      ...event,
-      duration: {
-        start: new Date(event.duration.start),
-        end: new Date(event.duration.end),
-      },
-    })
+    console.error("Could not find Original Event")
```
Comment (Victoria-Williamson) [inline utility/hooks/trip.tsx:371]: This won't do anything in production, maybe a different fallback would be to add the event locally if sometime happens in between rerenders and it cannot be found.

File: __mocks__/hooks/modifyEvent.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  title: "MODIFIED Sample Title",
+  duration: {
+    start: new Date("2023-04-03T23:04:45.155Z"),
+    end: new Date("2023-04-03T23:14:45.155Z"),
+  },
+  location: "MODIFIED Sample Location",
+  description: "MODIFIED Sample Description",
+  uid: "Sample uid",
+}
+export const MODIFIED_EVENT_WITH_ATTENDEES = {
```
Comment (Victoria-Williamson) [inline __mocks__/hooks/modifyEvent.tsx:24]: doesnt have attendees

**Review by Victoria-Williamson — CHANGES_REQUESTED** [review]

File: __mocks__/hooks/modifyEvent.tsx (hunk 7 above)
Comment (aamare1001) [reply __mocks__/hooks/modifyEvent.tsx:24]: Modified Events don't have Attendees due to the API not updating the attendees

File: utility/hooks/modify/modifyEvent.tsx [hunk 8]
```diff
@@ -1,10 +1,20 @@
-import React from "react"
+import React, { Dispatch, SetStateAction } from "react"
```
[the remark this answers is not in the document]
Comment (aamare1001) [reply utility/hooks/modify/modifyEvent.tsx:1]: setmodifiedevent but that's gone now

**Review by Victoria-Williamson — APPROVED** [review]
